src/main/java/com/ldtteam/structurize/client/BlueprintRenderer.java

```diff
@@ -173,6 +147,11 @@ private void init(final Blueprint blueprint, final Map<Object, Exception> suppre
                     state = Blocks.AIR.defaultBlockState();
                 }
             }
+            else if (Structurize.getConfig().getClient().renderPlaceholdersNice.get() && state.getBlock() == ModBlocks.blockSolidSubstitution.get()
+                && previewData.getSolidSubstitutionOverride() != null)
+            {
+                state = previewData.getSolidSubstitutionOverride();
+            }
```

Comment:
This should go into blockAccess so everyone can determine proper blockstate

Reply:
ye was thinking about that aswell, though we dont seem to have the preview data context in there. I could make an overload to pass it into it optionally

Reply:
nvm, I see we can just pass it along to the block access on init

Reply:
either pass or pass handle as function/supplier, not sure what is the memory context there, because you can have multiple previewData for single renderer

Reply:
updated the pr to it^^